language/move-lang/src/lib.rs

```diff
@@ -160,55 +160,43 @@ fn check_errors(errors: Errors) -> Result<(), Errors> {
 // Parsing
 //**************************************************************************************************
 
-fn parsing_error<Token>(
-    fname: &'static str,
-    e: lalrpop_util::ParseError<usize, Token, &'static str>,
-) -> Error
-where
-    Token: std::fmt::Display,
-{
+fn parsing_error(fname: &'static str, e: ParseError) -> Error {
     let fmt_expected = |expected: Vec<String>| -> String {
-        format!(
-            "Expected: {}",
-            expected
-                .iter()
-                .fold(String::new(), |acc, token| format!("{} {},", acc, token))
-        )
+        let mut msg = String::new();
+        for (i, e) in expected.iter().enumerate() {
+            let sep = match i {
```

Comment:
I appreciate the art in "or", but would like to point to the existence of [join](https://doc.rust-lang.org/std/primitive.slice.html#method.join)

Reply:
Yes, if we don't care about using a different separate between the last elements ("or"), it would make sense to use join here. I'm not a big fan of this particular error message regardless, but I was trying to improve the readability as much as I could.

Reply:
I changed to use join for now.